packages/@aws-cdk/aws-rds/lib/cluster.ts

```diff
@@ -560,8 +560,11 @@ export class DatabaseCluster extends DatabaseClusterNew {
    *
    * @param [automaticallyAfter=Duration.days(30)] Specifies the number of days after the previous rotation
    * before Secrets Manager triggers the next automatic rotation.
+   *
+   * @param [excludeCharacters=' ;+%{}`/"\\#\'@'] Specifies characters to not include in generated passwords.
+   * Defaults to the superset of characters which will break DMS endpoints and characters which cause problems in shell scripts.
    */
-  public addRotationSingleUser(automaticallyAfter?: Duration): secretsmanager.SecretRotation {
+  public addRotationSingleUser(automaticallyAfter?: Duration, excludedCharacters?: string): secretsmanager.SecretRotation {
```

Comment:
Can we actually add an interface here? This code:

```ts
instance.addRotationSingleUser(
  cdk.Duration.days(7), // automaticallyAfter defaults to 30 days
  'abc', // excludedCharacters defaults to ' ;+%{}`/"\\#\'@', the combination of characters that break DMS and that cause problems in shell scripts.
);
```

looks extremely cryptic to me; I'd rather see this:

```ts
instance.addRotationSingleUser(cdk.Duration.days(7), { // automaticallyAfter defaults to 30 days
  excludedCharacters: 'abc', // excludedCharacters defaults to ' ;+%{}`/"\\#\'@', the combination of characters that break DMS and that cause problems in shell scripts.
});
```

Reply:
Yeah, I didn't like it, but I didn't want to break anything. Adding options and not including the duration felt ugly, so...